### PR TITLE
Update for the more restricted span status codes.

### LINKED ISSUE #1700: Update allowed span status codes
See this recent spec PR:  https://github.com/open-telemetry/opentelemetry-specification/pull/966

### PR DESCRIPTION
**Pull request by jkwatson**
resolves #1700 

Note: I made the numeric codes match the proposed protobuf status codes.

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/trace/Status.java [hunk 1]
```diff
@@ -44,163 +43,18 @@
...
-    /**
-     * The service is currently unavailable. This is a most likely a transient condition and may be
-     * corrected by retrying with a backoff.
-     *
-     * <p>See litmus test above for deciding between FAILED_PRECONDITION, ABORTED, and UNAVAILABLE.
-     *
-     * @since 0.1.0
-     */
-    UNAVAILABLE(14),
+    OK(0),
```
Comment (anuraaga) [inline api/src/main/java/io/opentelemetry/trace/Status.java:51]: Do we need the numbers anymore?

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/trace/Status.java:51]: I'm honestly not sure. I noticed that in the Jaeger mapping, we directly just copy that code over as a status code. Is that correct? No idea, either. I'd prefer to get rid of the numeric codes, and the accessor for them, but we need to decide what to do with the existing mappings.

File: extensions/trace_utils/src/main/java/io/opentelemetry/extensions/trace/CurrentSpanUtils.java [hunk 2]
```diff
@@ -121,7 +121,7 @@ public V call() throws Exception {
 
   private static void setErrorStatus(Span span, Throwable t) {
     span.setStatus(
-        Status.UNKNOWN.withDescription(
+        Status.ERROR.withDescription(
```
Comment (iNikem) [inline extensions/trace_utils/src/main/java/io/opentelemetry/extensions/trace/CurrentSpanUtils.java:124]: I realised that I still an open question about this use case: https://github.com/open-telemetry/opentelemetry-specification/issues/1008

Comment (jkwatson) [reply extensions/trace_utils/src/main/java/io/opentelemetry/extensions/trace/CurrentSpanUtils.java:124]: I'd be happy to revisit this when the issue is clarified. Meanwhile, it does seem like approximately preserving the existing behavior is a good thing for now. And, as I commented on that issue, I think the intent with `recordException` is that it's an error, most commonly.

File: api/src/main/java/io/opentelemetry/trace/Status.java (hunk 1 above)
Comment (anuraaga) [reply api/src/main/java/io/opentelemetry/trace/Status.java:51]: Got it - we can change this latter anyways.

**Review by anuraaga — APPROVED** [review]

File: api/src/main/java/io/opentelemetry/trace/Status.java [hunk 3]
```diff
@@ -44,163 +43,18 @@
...
 
-    /**
-     * Unrecoverable data loss or corruption.
-     *
-     * @since 0.1.0
-     */
-    DATA_LOSS(15),
+    /** The default status. */
+    UNSET(17),
```
Comment (bogdandrutu) [inline api/src/main/java/io/opentelemetry/trace/Status.java:54]: How did we come with this numbers? See https://github.com/open-telemetry/opentelemetry-proto/pull/224/files and maybe we should follow the same numbers so translation to otlp is just direct mapping.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/trace/Status.java:54]: I made them up. I was planning on fixing once the proto change was merged.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/trace/Status.java:54]: I'll update this PR to match what is in the proto PR, now that it looks like there aren't any more changes happening.

Comment (anuraaga) [reply api/src/main/java/io/opentelemetry/trace/Status.java:54]: I think the numbers are now matching the deprecated code, but we should update to 0, 1, 2 here I believe.

Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/trace/Status.java:54]: yes. will do...that was my intent. :)

File: extensions/trace_utils/src/main/java/io/opentelemetry/extensions/trace/CurrentSpanUtils.java (hunk 2 above)
Comment (jkwatson) [reply extensions/trace_utils/src/main/java/io/opentelemetry/extensions/trace/CurrentSpanUtils.java:124]: When that gets merged, I'll create an issue to address it here. Meanwhile, keeping the existing behavior seems like the right decision for this PR.

File: api/src/main/java/io/opentelemetry/trace/Status.java (hunk 3 above)
Comment (jkwatson) [reply api/src/main/java/io/opentelemetry/trace/Status.java]: done

**Review by iNikem — APPROVED** [review]

**Pull request merged**